spec/Section 4 -- Composition.md

```diff
@@ -3179,6 +3179,127 @@ input BookFilter {
 In this invalid case, `title` is mandatory in Schema A but not defined in
 `Schema B`, causing inconsistency in required fields across schemas.
 
+#### Required Argument Missing
+
+**Error Code:**  
+
+`REQUIRED_ARGUMENT_MISSING_IN_SOME_SCHEMA`
+
+**Severity:**  
+
+ERROR
+
+**Formal Specification:**
+
+- Let {typeNames} be the set of all object and interface types names from all
+  source schemas that are not declared as `@internal`
+- For each {typeName} in {typeNames}:
+  - Let {typeDefinitions} be the list of all type definitions from different
+    source schemas with the name {typeName}.
+  - Let {fieldNames} be the set of all field names from all {typeDefinitions}
+    that are not declared as `@internal`.
+  - For each {fieldName} in {fieldNames}:
+    - Let {fieldDefinitions} be the list of all field definitions from
+      {typeDefinitions} with the name {fieldName}.
+    - Let {requiredArgumentNames} be the set of all argument names from
+      {fieldDefinitions} that have a non-nullable type in at least one
+      definition that does not specify `@require`
+    - For each {fieldDefinition} in {fieldDefinitions}:
+      - For each {requiredArgumentName} in {requiredArgumentNames}:
+        - {fieldDefinition} must have an argument with the name
+          {requiredArgumentName} that does not specify `@require`
+
+**Explanatory Text:**
+
+When merging a field definition across multiple schemas, any argument that is
+non-null (i.e., “required”) in one schema must appear in all schemas that define
+that field . In other words, arguments are effectively merged by intersection:
+if an argument is considered required in any schema, then that same argument
+must exist in every schema that contributes to the composite definition. If a
+required argument is missing in one schema, there is no consistent way to define
+that field or directive across schemas.
```

Comment:
The spec doesn't mention directive arguments?

Reply:
well spotted!